### PR TITLE
Le Improvment Work! :parrot: 

### PR DESCRIPTION
**Pull request by fliiiix**
Things this PR does:
- too much
- upgrading to poetry for the backend
- implementing a token claim system
- implementing the delete on top of the token system
- typos

Fixes: #49

### CODE REVIEW

**Comment by fliiiix** [comment]
image
@randombenj plz fix

File: docat/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+format:
```
Comment (randombenj) [inline docat/Makefile:1]: Do we need he makefile if we dont use it in e.g. docke

File: docat/app.py [hunk 2]
```diff
@@ -9,17 +9,19 @@
...
+from docat.docat.utils import UPLOAD_FOLDER, create_nginx_config, create_symlink, extract_archive, remove_docs
 
 app = Flask(__name__)
-app.config["UPLOAD_FOLDER"] = os.getenv("DOCAT_DOC_PATH", "/var/docat/doc")
+app.config["UPLOAD_FOLDER"] = os.getenv("DOCAT_DOC_PATH", UPLOAD_FOLDER)
 app.config["MAX_CONTENT_LENGTH"] = 100 * 1024 * 1024  # 100M
+app.db = TinyDB('db.json')
 
 
 @app.route("/api/<project>/<version>", methods=["POST"])
```
Comment (randombenj) [inline docat/app.py:27]: Schould we autoclaim on the first uppload? Or is this not obvious enough?

Comment (randombenj) [reply docat/app.py:27]: Also should we also add (maybe optional) token authentication for upploads?

File: docat/app.py [hunk 3]
```diff
@@ -50,7 +52,7 @@ def upload(project, version):
 @app.route("/api/<project>/<version>/tags/<new_tag>", methods=["PUT"])
```
Comment (randombenj) [inline docat/app.py:52]: Also here optional token auth?

File: docat/Makefile (hunk 1 above)
Comment (fliiiix) [reply docat/Makefile:1]: yes.

File: docat/app.py (hunk 2 above)
Comment (fliiiix) [reply docat/app.py:27]: we could add a protected mode where you need the token to change anything - but i would not add it by default

File: docat/Makefile (hunk 1 above)
Comment (randombenj) [reply docat/Makefile:1]: For wat?

**Comment by randombenj** [comment]
Two more things (sorry, not sorry). What do you think of hashing the keys before storing to the db? And I think claim/delete should be documented in the readme 😇

File: docat/Makefile (hunk 1 above)
Comment (fliiiix) [reply docat/Makefile:1]: fun and because i am lazy

**Comment by fliiiix** [comment]
Ok let me know what you think I added the tokens hashed and some docs

**Pull request merged**